pihole

```diff
@@ -218,6 +225,65 @@ tailFunc() {
   exit 0
 }
 
+piholeCheckoutFunc() {
+  source "${PI_HOLE_SCRIPT_DIR}"/piholeCheckout.sh
+  local corebranches
+  local webbranches
+
+  # Avoid globbing
+  set -f
+
+  #This is unlikely
+  if ! is_repo "${PI_HOLE_FILES_DIR}" || ! is_repo "${ADMIN_INTERFACE_DIR}" ; then
+    echo "::: Critical Error: One or more Pi-Hole repos are missing from system!"
+    echo "::: Please re-run install script from https://github.com/pi-hole/pi-hole"
+    exit 1
+  fi
+
+  if ! warning1 ; then
+    echo "::: Not confirmed. Back to safety"
+    exit 1
+  fi
+
+  echo -n "::: Fetching remote branches for Pi-hole core from ${PI_HOLE_GIT_URL} ... "
+#  fully_fetch_repo ${PI_HOLE_FILES_DIR}
+  corebranches=($(get_available_branches "${PI_HOLE_FILES_DIR}"))
+  echo " done!"
+  echo "::: ${#corebranches[@]} branches available"
+  echo ":::"
+
+  echo -n "::: Fetching remote branches for the web interface from ${ADMIN_INTERFACE_GIT_URL} ... "
+#  fully_fetch_repo ${ADMIN_INTERFACE_DIR}
+  webbranches=($(get_available_branches "${ADMIN_INTERFACE_DIR}"))
+  echo " done!"
+  echo "::: ${#webbranches[@]} branches available"
+  echo ":::"
+
+  if [[ "${2}" == "dev" ]] ; then
+    # Shortcut to check out development version
+    echo "::: Shortcut \"dev\" detected - checking out development / devel branches ..."
+    echo "::: Pi-hole core"
+    checkout_branch "${PI_HOLE_FILES_DIR}" "development"
+    echo "::: Web interface"
+    checkout_branch "${ADMIN_INTERFACE_DIR}" "devel"
+    echo "::: done!"
+  else
+    # Have to user chosing the branch he wants
+    select_branch "${corebranches[@]}"
+    corebranch=$(cat /etc/pihole/checkout-choice)
+    select_branch "${webbranches[@]}"
+    webbranch=$(cat /etc/pihole/checkout-choice)
```

Comment:
Bash doesn't return non-numerics very well, but I'd check here for an empty file (which would mean a cancelled whiptail.)